### PR TITLE
openjdk: 10.0.2 -> 11.0.1

### PR DESCRIPTION
**Pull request by jerith666**
###### Major TODOs:

- [x] JRE is deprecated, they recommend using `jmod`
- [x] ARM doesn't build openjdk but uses oraclejdk
   - ~get them building~
      - ~need bootstrap jdk for ARM~
   - ~update oraclejdk to 11 too~
   - [x] realize that 10 wasn't working on ARM anyway (#48649)
- [ ] Consider reusing some of the AdoptOpenJDK build scripts? https://github.com/AdoptOpenJDK/openjdk-build
- [x] Update darwin

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by volth** [comment]
It probably fixes https://github.com/NixOS/nixpkgs/issues/37364

**Comment by jerith666** [comment]
We'll need a bootstrap jdk for arm before we can build openjdk for that architecture.

**Comment by jerith666** [comment]
Squashed some of my WIP comments ... I think I need someone who has access to darwin to make that update, though?

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -8098,10 +8095,10 @@ with pkgs;
     inherit (darwin) cctools;
     inherit (darwin.apple_sdk.frameworks) CoreFoundation CoreServices Foundation;
   };
-  bazel_jdk10 = callPackage ../development/tools/build-managers/bazel {
+  bazel_jdk11 = callPackage ../development/tools/build-managers/bazel {
```
Comment (uri-canva) [inline pkgs/top-level/all-packages.nix]: 👍

**Comment by jerith666** [comment]
Okay, I've tested that eclipse runs with jdk11 locally.  I think once someone can confirm that the darwin jdk11 is basically working, this is ready to merge!

**Comment by uri-canva** [comment]
Tested this to use Bazel with jdk11 on darwin, failed:
```
these derivations will be built:
  /nix/store/0vw2fndb535kqj1bbfp42h26zy5pxjv6-zulu11.2.3-jdk11.0.1-linux_x64.tar.gz.drv
  /nix/store/d7bq5ir7mxhs87h0zy88daa1g5glkyjl-zulu11.2.3-jdk11.0.1.drv
  /nix/store/3fb0mnlp8b680ywdl3gvggs02vzcn3py-bazel-0.17.1.drv
these paths will be fetched (73.15 MiB download, 76.57 MiB unpacked):
  /nix/store/08hdv1dsr6ngmyq47h25nqcg3k4fdfmb-bazel-deps
  /nix/store/rd9bdnf17wsfqclyqz4gv3gfb58nfsx8-bazel-0.17.1-dist.zip
  /nix/store/xfc7wancf1x9m0qnc319dfkq1l9viq49-27be70979b54d7510bf401d9581fb4075737ef34.patch
copying path '/nix/store/xfc7wancf1x9m0qnc319dfkq1l9viq49-27be70979b54d7510bf401d9581fb4075737ef34.patch' from 'https://cache.nixos.org'...
copying path '/nix/store/rd9bdnf17wsfqclyqz4gv3gfb58nfsx8-bazel-0.17.1-dist.zip' from 'https://cache.nixos.org'...
copying path '/nix/store/08hdv1dsr6ngmyq47h25nqcg3k4fdfmb-bazel-deps' from 'https://cache.nixos.org'...
building '/nix/store/0vw2fndb535kqj1bbfp42h26zy5pxjv6-zulu11.2.3-jdk11.0.1-linux_x64.tar.gz.drv'...

trying https://cdn.azul.com/zulu/bin/zulu11.2.3-jdk11.0.1-linux_x64.tar.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  187M  100  187M    0     0  5955k      0  0:00:32  0:00:32 --:--:-- 8910k
building '/nix/store/d7bq5ir7mxhs87h0zy88daa1g5glkyjl-zulu11.2.3-jdk11.0.1.drv'...
unpacking sources
unpacking source archive /nix/store/az6zgzvxqs53ns9pvd8bmhcjc68plrhy-zulu11.2.3-jdk11.0.1-linux_x64.tar.gz
source root is zulu11.2.3-jdk11.0.1-linux_x64
setting SOURCE_DATE_EPOCH to timestamp 1539936296 of file zulu11.2.3-jdk11.0.1-linux_x64/readme.txt
patching sources
configuring
no configure script, doing nothing
building
no Makefile, doing nothing
installing
Archive:  /nix/store/xkf1p54gf3igqr3qvwxgrrk4asfxl4qp-ZuluJCEPolicies.zip
   creating: ZuluJCEPolicies/
  inflating: ZuluJCEPolicies/LICENSE
  inflating: ZuluJCEPolicies/README.txt
  inflating: ZuluJCEPolicies/DISCLAIMER
  inflating: ZuluJCEPolicies/local_policy.jar
  inflating: ZuluJCEPolicies/US_export_policy.jar
post-installation fixup
error: install_name_tool: can't open file: /nix/store/knc10wvsrgca7schl7d6hpplhvrsbwg4-zulu11.2.3-jdk11.0.1/lib/libfontmanager.dylib (No such file or directory)
builder for '/nix/store/d7bq5ir7mxhs87h0zy88daa1g5glkyjl-zulu11.2.3-jdk11.0.1.drv' failed with exit code 1
error: build of '/nix/store/3fb0mnlp8b680ywdl3gvggs02vzcn3py-bazel-0.17.1.drv', '/nix/store/d7bq5ir7mxhs87h0zy88daa1g5glkyjl-zulu11.2.3-jdk11.0.1.drv' failed
```
It seems like it's using the Linux derivation here:
```
$ ls -l /nix/store/knc10wvsrgca7schl7d6hpplhvrsbwg4-zulu11.2.3-jdk11.0.1/lib/
total 410608
-rw-r--r--  1 uri  staff      39427 19 Oct 18:55 classlist
-rw-r--r--  1 uri  staff    4489118 19 Oct 18:55 ct.sym
-rwxr-xr-x  1 uri  staff       9863 19 Oct 18:55 jexec
drwxr-xr-x  4 uri  staff        128 19 Oct 18:55 jfr
drwxr-xr-x  3 uri  staff         96 19 Oct 19:07 jli
-rw-r--r--  1 uri  staff     106630 19 Oct 18:55 jrt-fs.jar
-rw-r--r--  1 uri  staff         29 19 Oct 18:55 jvm.cfg
-rw-r--r--  1 uri  staff      12769 19 Oct 18:55 libattach.so
-rw-r--r--  1 uri  staff     761905 19 Oct 18:55 libawt.so
-rw-r--r--  1 uri  staff      42709 19 Oct 18:55 libawt_headless.so
-rw-r--r--  1 uri  staff     490199 19 Oct 18:55 libawt_xawt.so
-rw-r--r--  1 uri  staff      25443 19 Oct 18:55 libdt_socket.so
-rw-r--r--  1 uri  staff      11432 19 Oct 18:55 libextnet.so
-rw-r--r--  1 uri  staff     842211 19 Oct 18:55 libfontmanager.so
-rw-r--r--  1 uri  staff     772361 19 Oct 18:55 libfreetype.so
-rw-r--r--  1 uri  staff      49510 19 Oct 18:55 libinstrument.so
-rw-r--r--  1 uri  staff      46995 19 Oct 18:55 libj2gss.so
-rw-r--r--  1 uri  staff      15994 19 Oct 18:55 libj2pcsc.so
-rw-r--r--  1 uri  staff      80337 19 Oct 18:55 libj2pkcs11.so
-rw-r--r--  1 uri  staff       8170 19 Oct 18:55 libjaas.so
-rw-r--r--  1 uri  staff     225489 19 Oct 18:55 libjava.so
-rw-r--r--  1 uri  staff     247117 19 Oct 18:55 libjavajpeg.so
-rw-r--r--  1 uri  staff       7803 19 Oct 18:55 libjawt.so
-rw-r--r--  1 uri  staff     276208 19 Oct 18:55 libjdwp.so
-rw-r--r--  1 uri  staff     135284 19 Oct 18:55 libjimage.so
-rw-r--r--  1 uri  staff      14366 19 Oct 18:55 libjsig.so
-rw-r--r--  1 uri  staff      82128 19 Oct 18:55 libjsound.so
-rw-r--r--  1 uri  staff     505031 19 Oct 18:55 liblcms.so
-rw-r--r--  1 uri  staff      25895 19 Oct 18:55 libmanagement.so
-rw-r--r--  1 uri  staff       7751 19 Oct 18:55 libmanagement_agent.so
-rw-r--r--  1 uri  staff      30570 19 Oct 18:55 libmanagement_ext.so
-rw-r--r--  1 uri  staff     531075 19 Oct 18:55 libmlib_image.so
-rw-r--r--  1 uri  staff     113129 19 Oct 18:55 libnet.so
-rw-r--r--  1 uri  staff      86553 19 Oct 18:55 libnio.so
-rw-r--r--  1 uri  staff       9283 19 Oct 18:55 libprefs.so
-rw-r--r--  1 uri  staff       6781 19 Oct 18:55 librmi.so
-rw-r--r--  1 uri  staff      58795 19 Oct 18:55 libsaproc.so
-rw-r--r--  1 uri  staff      28671 19 Oct 18:55 libsctp.so
-rw-r--r--  1 uri  staff     370420 19 Oct 18:55 libsplashscreen.so
-rw-r--r--  1 uri  staff     318147 19 Oct 18:55 libsunec.so
-rw-r--r--  1 uri  staff     217017 19 Oct 18:55 libunpack.so
-rw-r--r--  1 uri  staff      64725 19 Oct 18:55 libverify.so
-rw-r--r--  1 uri  staff      37209 19 Oct 18:55 libzip.so
-rw-r--r--  1 uri  staff  140987885 19 Oct 18:55 modules
-rw-r--r--  1 uri  staff       3793 19 Oct 18:55 psfont.properties.ja
-rw-r--r--  1 uri  staff      11390 19 Oct 18:55 psfontj2d.properties
drwxr-xr-x  8 uri  staff        256 29 Oct 11:01 security
drwxr-xr-x  5 uri  staff        160 19 Oct 19:07 server
-rw-r--r--  1 uri  staff   57825731 19 Oct 18:55 src.zip
-rw-r--r--  1 uri  staff     105882 19 Oct 18:55 tzdb.dat
```
They're Linux shared libraries:
```
$ file /nix/store/knc10wvsrgca7schl7d6hpplhvrsbwg4-zulu11.2.3-jdk11.0.1/lib/libfontmanager.so
/nix/store/knc10wvsrgca7schl7d6hpplhvrsbwg4-zulu11.2.3-jdk11.0.1/lib/libfontmanager.so: ELF 64-bit LSB shared object x86-64, version 1 (SYSV), dynamically linked, BuildID[sha1]=79c4d8cf0e7d1dd5c0dbc745bda6b4b70256dc9a, not stripped
```
As opposed to Darwin shared libraries:
```
$ file /usr/lib/libSystem.dylib
/usr/lib/libSystem.dylib: Mach-O universal binary with 2 architectures: [x86_64:Mach-O 64-bit dynamically linked shared library x86_64] [i386:Mach-O dynamically linked shared library i386]
/usr/lib/libSystem.dylib (for architecture x86_64):     Mach-O 64-bit dynamically linked shared library x86_64
/usr/lib/libSystem.dylib (for architecture i386):       Mach-O dynamically linked shared library i386
```

**Comment by jerith666** [comment]
Oh geeze -- facepalm!  I just followed the `Referer:https://www.azul.com/downloads/zulu/zulu-linux/` link, which points to linux, and cut-n-pasted away.  :(

Fixed in https://github.com/NixOS/nixpkgs/pull/48326/commits/1bd7b98c7975136ddd6183a155e90688fd5b3e43, please try again!

**Comment by uri-canva** [comment]
All good, `bazel_jdk11` builds and runs correctly with `1bd7b98c7975136ddd6183a155e90688fd5b3e43`. I wasn't able to test building java 11 code with it because your branch doesn't have `bazel-0.18.0` from current master, but only `bazel-0.17.1` from an older point, but I think it should be ok as far as the jdk is concerned.

**Review by fpletz — APPROVED** [review]
LGTM

**Pull request merged**